modules/standard/FileSystem.chpl

```diff
@@ -16,17 +16,78 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+/* A file utilities library
+
+   The FileSystem module focuses on file and directory properties and
+   operations.  It does not cover every interaction involving a file -
+   for instance, path specific operations live in the :mod:`Path` module, while
+   opening, writing to, or reading from a file appropriately live in the
+   :mod:`IO` module.  Rather, it covers cases where the user would prefer a file
+   or directory handled wholesale and/or with minimal interaction.  The
+   :proc:`copy` functionality belongs here, as does :proc:`chmod`,
+   :proc:`mkdir`, and :proc:`remove`, for instance.  Also included are
+   operations relating to the current process's state, such as :proc:`umask` or
+   :proc:`~locale.chdir` (an operation which is performed on a specified
+   locale).
+
+ */
+module FileSystem {
+
 use Error;
 
+/* S_IRUSR and the following constants are values of the form
+   S_I[R | W | X][USR | GRP | OTH], S_IRWX[U | G | O], S_ISUID, S_ISGID, or
+   S_ISVTX, where R corresponds to readable, W corresponds to writable, X
+   corresponds to executable, USR and U correspond to user, GRP and G
+   correspond to group, OTH and O correspond to other, directly tied to the C
+   idea of these constants.  They are intended for use when dealing with the
+   permissions of files or directories, such as with :proc:`chmod`,
+   :proc:`getMode`, :proc:`mkdir`, or :proc:`umask`
 
-/* Change the current working directory of the current locale to the specified
-   name. Returns any errors that occurred via an out parameter.
-   err: a syserr used to indicate if an error occurred
-   name: a string indicating a new directory
+   S_IRUSR refers to the user's read permission
+*/
+extern const S_IRUSR: int;
+/* Refers to the user's write permission, see :data:`S_IRUSR` */
+extern const S_IWUSR: int;
+/* Refers to the user's executable permission, see :data:`S_IRUSR` */
+extern const S_IXUSR: int;
+/* Refers to all the user's permissions, see :data:`S_IRUSR` */
```

Comment:
I'd like this and the other similar statements to be more clear about "all the user's permissions" by saying something like "all the user's permissions - read, write, and execute".